### PR TITLE
PF4: Page component managed sidebar

### PR DESCRIPTION
**Pull request by jschuler**
Closes: #913 

Add new prop to Page component isManagedSidebar which handles the sidebar toggling by maintaining a separate state for "desktop view" and "mobile view".

Use can still manage the sidebar toggle themselves if desired, and can implement this behavior by adding a callback to the new onPageResize prop on the Page component.

@matthewcarleton

### CODE REVIEW

**Comment by jschuler** [comment]
Examples here: https://1249-pr-patternfly-react-patternfly.surge.sh/patternfly-4/demos/pagelayout

File: packages/patternfly-4/react-core/src/components/Page/Page.js [hunk 1]
```diff
@@ -25,21 +33,80 @@ const defaultProps = {
...
-        {children}
-      </main>
-    </div>
-  );
-};
+class Page extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
Comment (ibolton336) [inline packages/patternfly-4/react-core/src/components/Page/Page.js]: Would be a good opportunity to switch this component to class properties & get rid of the constructor:

https://codeburst.io/use-class-properties-to-clean-up-your-classes-and-react-components-93185879f688

**Comment by christianvogt** [comment]
@jschuler a typical pattern in react for differentiating between controlled and uncontrolled state is that the component is uncontrolled (handles the state internally) if the prop which controls the component is not set. Once that prop is set, the component becomes controlled. Then to provide a default value for the uncontrolled state, you prefix the control prop with `default`.

In this case I would say that `isManagedSidebar` is not necessary. If `isNavOpen` is not set, then the component is uncontrolled and its open state is managed internally. To provide a default value (not that it makes sense for the component though, but as an example), the prop `defaultIsNavOpen` can be provided and still have the component be uncontrolled.

**Comment by jschuler** [comment]
Thanks @christianvogt, that makes a lot of sense. Have opened an issue to follow up on this.

**Pull request merged**